uwazi/entity-page/src/lib/Settings.svelte

```diff
@@ -0,0 +1,222 @@
+    // Don't save jwt value
+    jwt = "";
+  });
+
+  $: if (success) {
+    // Reload data if sources are actually modified due to request success
+    sourcesCopy = structuredClone($hyperbeeSources);
+    sourcesBackup = structuredClone($hyperbeeSources);
+  }
+</script>
+
+<h2>Sources</h2>
+{#if success === false}
+  <p>Sources will not be updated due to error, edit and try to save again.</p>
+{/if}
+{#if success === null}
+  <p>Unsaved changes</p>
+{/if}
+
+<div id="sources-list" class="section">
+  {#each sourcesCopy as { name, server }, i}
+    <div class="source input-style">
+      <!-- svelte-ignore a11y-click-events-have-key-events -->
+      <svg
+        on:click={() => {
+          success = null;
+          sourcesCopy.splice(i + 1, 0, {
+            name: "",
+            server: "",
+          });
+          sourcesCopy = sourcesCopy; // Reactivity
+        }}
+        class="icon plus"
+        xmlns="http://www.w3.org/2000/svg"
+        viewBox="0 0 448 512"
+        ><!--! Font Awesome Pro 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) Copyright 2023 Fonticons, Inc. --><path
+          d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
+        /></svg
+      >
+      <div class="input-container">
+        <input
+          class="name"
+          type="text"
+          placeholder="ACME Corp News"
+          bind:value={name}
+          on:input={() => (success = null)}
+        />
+        <input
+          class="server"
+          type="url"
+          placeholder="https://example.com"
+          bind:value={server}
+          on:input={() => (success = null)}
+        />
+      </div>
+      <!-- svelte-ignore a11y-click-events-have-key-events -->
+      <svg
+        on:click={() => {
+          success = null;
+          sourcesCopy.splice(i, 1);
+          sourcesCopy = sourcesCopy; // Reactivity
+        }}
+        class="icon trash"
+        xmlns="http://www.w3.org/2000/svg"
+        viewBox="0 0 448 512"
+        ><!--! Font Awesome Pro 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) Copyright 2023 Fonticons, Inc. --><path
+          d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0H284.2c12.1 0 23.2 6.8 28.6 17.7L320 32h96c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 96 0 81.7 0 64S14.3 32 32 32h96l7.2-14.3zM32 128H416V448c0 35.3-28.7 64-64 64H96c-35.3 0-64-28.7-64-64V128zm96 64c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16z"
+        /></svg
+      >
+    </div>
+  {/each}
+</div>
+<h2>Login</h2>
+<p>Enter JWT to allow for writing new data.</p>
```

Comment:
Per-source is definitely an option, I was trying to keep the initial implementation simple. The use case I was imagining that seems most likely to me is a user that subscribes to several sources, but only has one hyperbee that they actually control, a personal one, or one for their organization. In that case only one JWT is needed.

Reply:
Ok, it's definitely simpler to only have one login form at least for demo purposes. However, I think to prevent confusion, the login form should be labeled with the source url so we know which source we are logging into

Reply:
The way things work right now, the JWT is actually not tied to any source. It's used for whatever source you select when you click the copy or save buttons. I guess should change this, as accidentally sending your auth info to another server is not a good idea. Will do.

Reply:
Ok now in 112eb1766673671dfb7606f68b55d93d7caf0d87 the JWT is only sent the first source. Let me know if that works.

Reply:
looks good!